server/pages/index.jade

```diff
@@ -110,25 +110,45 @@ html(lang=lang, dir=isRTL ? 'rtl' : 'ltr', class=isFluidWidth ? 'is-fluid-width'
 				}
 			})();
 
-		if user
+		if supportUser
+			script.
+				window.supportUser = {
+					readyScripts: []
+				};
+		if user && !supportUser
 			script(type='text/javascript')!='var currentUser = ' + sanitize.jsonStringifyForHtml( user )
 		if app
 			script(type='text/javascript')!='var app = ' + sanitize.jsonStringifyForHtml( app )
-		if initialReduxState
+		if initialReduxState && !supportUser
```

Comment:
@seear: Ah, thanks for the insight. That makes sense - I'll move the logic into the initial context setup.

@mtias: We need to make sure that no original user information is left in the page, or we'll end up with inconsistent data - some that belongs to the original user, some that belongs to the supported user. There is unfortunately a fair bit of complexity introduced by this feature; it's like adding multi-user support to single-user software :smile: I think moving these checks to the server side context setup should tidy this up though.

Let's move this discussion to #4068 as it explains the reasoning more clearly and in detail without the localForage changes.